Common/ObjectSerializer.cs

```diff
@@ -1,16 +1,17 @@
 using System.Runtime.Serialization;
 using System.Runtime.Serialization.Formatters.Binary;
 using Common.Enum;
+using Common.Interfaces;
 
 namespace Common;
 
-public class ObjectSerializer
+public class ObjectSerializer : IObjectSerializer
 {
     public string Location { get; set; }
     public ILogger Logger { get; set; }
-    public ObjectSerializer(string location, ILogger logger)
+    public ObjectSerializer(ILogger logger)
     {
-        Location = $"{location}/cache";
+        Location = EnvironmentVariables.GetVariable("location") + "/cache";
         bool exists = System.IO.Directory.Exists(Location);
         if (!exists)
             System.IO.Directory.CreateDirectory(Location);
```

Comment:
Maybe create a fileHandler or fileManager to do the file handling logic and get it out of the serializer class ( seperation of concerns )

Reply:
#65